src/fibsem_tools/io/h5.py

```diff
@@ -7,78 +7,116 @@
-                     "track_order",
-                     "external",
-                     "allow_unknown_filter")
-
-H5_GROUP_KWDS = ("name",
-                "track_order")
-
-H5_FILE_KWDS = ("name",
-                  "mode",
-                  "driver",
-                  "libver",
-                  "userblock_size",
-                  "swmr",
-                  "rdcc_nslots",
-                  "rdcc_nbytes",
-                  "rdcc_w0",
-                  "track_order",
-                  "fs_strategy",
-                  "fs_persist",
-                  "fs_threshold")
+H5_DATASET_KWDS = (
+    "name",
+    "shape",
+    "dtype",
+    "data",
+    "chunks",
+    "compression",
+    "compression_opts",
+    "scaleoffset",
+    "shuffle",
+    "fletcher32",
+    "maxshape",
+    "fillvalue",
+    "track_times",
+    "track_order",
+    "dcpl",
+    "external",
+    "allow_unknown_filter",
+)
+
+H5_GROUP_KWDS = ("name", "track_order")
+
+H5_FILE_KWDS = (
+    "name",
+    "mode",
+    "driver",
+    "libver",
+    "userblock_size",
+    "swmr",
+    "rdcc_nslots",
+    "rdcc_nbytes",
+    "rdcc_w0",
+    "track_order",
+    "fs_strategy",
+    "fs_persist",
+    "fs_threshold",
+)
+
+
+# Could use multiple inheritance here
+class ManagedDataset(h5py.Dataset):
+    """
+    h5py.Dataset with context manager behavior
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, ex_type, ex_value, ex_traceback):
+        self.file.close()
+
+
+class ManagedGroup(h5py.Group):
+    """
+    h5py.Group with context manager behavior
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, ex_type, ex_value, ex_traceback):
+        self.file.close()
+
 
 def partition_h5_kwargs(**kwargs) -> Tuple[Dict[str, Any], Dict[str, Any]]:
     """
     partition kwargs into file-creation kwargs and dataset-creation kwargs
     """
     file_kwargs = kwargs.copy()
     dataset_kwargs = {}
-    for key in H5_DATASET_KWDS:
-        if key in file_kwargs:
+    for key in kwargs:
+        if key in H5_DATASET_KWDS:
             dataset_kwargs[key] = file_kwargs.pop(key)
 
     return file_kwargs, dataset_kwargs
 
 
 def access_h5(
-    store: Pathlike, path: Pathlike, mode: str, **kwargs
+    store: Union[h5py.File, Pathlike], path: Pathlike, **kwargs
 ) -> Union[h5py.Dataset, h5py.Group]:
     """
     Docstring
     """
-    if mode not in H5_ACCESS_MODES:
-        raise ValueError(f"Invalid access mode. Got {mode}, expected one of {H5_ACCESS_MODES}.")
-
     attrs = kwargs.pop("attrs", {})
+    mode = kwargs.get("mode", "r")
     file_kwargs, dataset_kwargs = partition_h5_kwargs(**kwargs)
-    
-    h5f = h5py.File(store, mode=mode, **file_kwargs)
 
-    if mode in ("r", "r+", "a") and (result := h5f.get(path)) is not None:
-        return result
+    if isinstance(store, h5py.File):
+        h5f = store
+    else:
+        h5f = h5py.File(store, **file_kwargs)
+
+    if mode in ("r", "r+", "a"):
```

Comment:
Perhaps if path is empty, just return the File, `h5f`.

Reply:
If we are in `r+` or `a` mode, perhaps the path does not exist because we are trying to create it. In those cases, perhaps you should catch the error.